File: 2075.cpp
struct Solution {
    inline string decodeCiphertext(string &encodedText, int rows) {
        int n = encodedText.size();
        if (rows == 1 || n == 0){
            return encodedText;
        }
        int rowlen = n / rows;
        string res;
        for (int i = 0; i < rowlen; ++i){
            for (int j = i; j < n; j += rowlen + 1){
                res.push_back(encodedText[j]);
            }
        }
        while (res.back() == ' '){
            res.pop_back();
        }
        return res;
    }
};

File: 2075.py
class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        if rows == 1 or len(encodedText) == 0:
            return encodedText
        n = len(encodedText)
        rowlen = n // rows
        return ''.join(encodedText[j::rowlen + 1] for j in range(rowlen)).rstrip()
